iotwifi/wpacfg.go

```diff
@@ -114,17 +116,49 @@ rsn_pairwise=CCMP`
 			wpa.Log.Info("Hostapd DISABLED")
 			//cmd.Process.Kill()
 			//cmd.Wait()
-
-			return
-
+			return cmd
 		}
 		if strings.Contains(out, "uap0: AP-ENABLED") {
 			wpa.Log.Info("Hostapd ENABLED")
-			return
+			return cmd
 		}
 	}
 }
 
+// Status returns the AP status.
+func (wpa *WpaCfg) APStatus() (map[string]interface{}, error) {
+	cfgMap := make(map[string]interface{}, 0)
+
+	// get the standard stats
+	stateOut, err := exec.Command("hostapd_cli", "-i", "uap0", "status").Output()
+	if err != nil {
+		wpa.Log.Fatal("Got error checking state: %s", err.Error())
+		return cfgMap, err
+	}
+
+	// Remove the indexing associated with ssid, bssid, and bss
+	stateOut = bytes.ReplaceAll(stateOut, []byte("[0]"), []byte(""))
+	cfgMap = cfgMapper(stateOut)
+
+	// get the list of connected clients
+	clientsOut, err := exec.Command("hostapd_cli", "-i", "uap0", "list_sta").Output()
```

Comment:
This returns a list of `\n` separated mac address strings representing clients that are connected to the AP.

Reply:
There's a good stdlib `strings` package which makes this easier. I'm indifferent here but since we need it in a string form anyway it might be easier to do something like:

```
lines := range strings.Split(string(clientsOut), "\n")
```

Reply:
Makes sense.  We need to do the conversion somewhere - seems more reasonable to do it once on the exec.Command.  I was mimicking the pattern below in the cfgMapper func.

Reply:
As I dig into this more I think it makes sense to let the cfgMapper function do the bytes to string conversion as it's built to handle the raw output from the cli's.  I'll convert the other areas to process as strings asap.